app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: int, age: str) -> None:
+        self.name = name
+        self.age = age
+        self.spouse = None
+        Person.people[name] = self
+
+    def set_spouse(self, spouse_name: int) -> None:
+        if spouse_name in Person.people:
+            self.spouse = Person[spouse_name]
+            Person.people[spouse_name].spouse = self
+
+    def create_person_list(self, people: list) -> list:
+        person_instances = []
+
+        for person_info in people:
+            name = person_info["name"]
+            age = person_info["age"]
+            person = Person(name, age)
+            spouse_name = person_info.get("wife") or person_info.get("husband")
+            if spouse_name:
+                person.set_spouse(spouse_name)
+            person_instances.append(person)
```

Comment:
You will need to make two iterations - the first one for creating all `People` instances, the second one for setting their spouses so all people will be created upon that time.